app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_catcher/stream.rb

```diff
@@ -17,32 +15,21 @@ def stop
     @stop_polling = true
   end
 
-  def fake_events
-    [
-      OpenStruct.new(:name       => %w[instance_power_on instance_power_off].sample,
-                     :id         => Time.zone.now.to_i,
-                     :timestamp  => Time.zone.now,
-                     :vm_ems_ref => [1, 2].sample),
-      OpenStruct.new(:name       => %w[instance_power_on instance_power_off].sample,
-                     :id         => Time.zone.now.to_i + 1,
-                     :timestamp  => Time.zone.now,
-                     :vm_ems_ref => [1, 2].sample),
-    ]
-  end
-
   def poll
-    @ems.with_provider_connection do |_connection|
-      catch(:stop_polling) do
-        loop do
-          fake_events.each do |activity|
-            throw :stop_polling if @stop_polling
-            yield activity.to_h
-          end
-          sleep @poll_sleep
-        end
-      rescue => exception
-        raise ProviderUnreachable, exception.message
+    from_time = Time.now.utc.to_i
+
+    loop do
+      ems.with_provider_connection(:service => 'PCloudEventsApi') do |api|
```

Comment:
Might be more efficient to setup a connection outside of the loop rather than re-connect every time:
```
pcloud_events_api = ems.connect(:service => "PCloudEventsApi")
loop do
  events = pcloud_events_api.....
end
```

You might need to rescue a stale bearer token and retry but it should reduce the number of API calls to the IAM service

Reply:
I refactored per your suggestion, but still need to lookup the correct Exception to rescue for a stale token (and test).

Reply:
My guess is `IbmCloudPower::ApiError` with `err.code => 401` but yeah, good to test it to confirm